lib/devices.go

```diff
@@ -119,7 +121,7 @@ func getDevices() (time.Time, []Device) {
 	ctx := gousb.NewContext()
 	defer func() {
 		if err := ctx.Close(); err != nil {
-			log.Printf("error closing USB context: %v\n", err)
+			addErrorLog(err.Error(), time.Now(), StateError)
```

Comment:
Did you check what the `err.Error()` messages are like? Are they properly human readable as-is or should you wrap them in something else?